src/app/clusters/time-synchronization-server/time-synchronization-cluster.h

```diff
@@ -123,33 +128,65 @@ class TimeSynchronizationServer : public FabricTable::Delegate
     // ReadClient::Callback functions
     void OnAttributeData(const ConcreteDataAttributePath & aPath, TLV::TLVReader * apData, const StatusIB & aStatus) override;
     void OnDone(ReadClient * apReadClient) override;
-
 #endif
 
     CHIP_ERROR AttemptToGetTimeFromTrustedNode();
 
     // Platform event handler functions
     void OnPlatformEventFn(const DeviceLayer::ChipDeviceEvent & event);
-
-    void OnTimeSyncCompletionFn(TimeSourceEnum timeSource, GranularityEnum granularity);
+    void OnTimeSyncCompletionFn(TimeSynchronization::TimeSourceEnum timeSource, TimeSynchronization::GranularityEnum granularity);
     void OnFallbackNTPCompletionFn(bool timeSyncSuccessful);
 
+    DataModel::ActionReturnStatus ReadAttribute(const DataModel::ReadAttributeRequest & request,
+                                                AttributeValueEncoder & encoder) override;
+    CHIP_ERROR Attributes(const ConcreteClusterPath & path, ReadOnlyBufferBuilder<DataModel::AttributeEntry> & builder) override;
+
+    std::optional<DataModel::ActionReturnStatus> InvokeCommand(const DataModel::InvokeRequest & request,
+                                                               TLV::TLVReader & input_arguments, CommandHandler * handler) override;
+
+    CHIP_ERROR AcceptedCommands(const ConcreteClusterPath & path,
+                                ReadOnlyBufferBuilder<DataModel::AcceptedCommandEntry> & builder) override;
+
+    CHIP_ERROR GeneratedCommands(const ConcreteClusterPath & path, ReadOnlyBufferBuilder<CommandId> & builder) override;
+
+    static TimeSynchronization::Delegate * GetDelegate()
```

Comment:
Having these static feels off ... this makes mDelegate as a member also be static and then the API surface is odd. 

This needs explanation and this feels like a workaround probably for a `delegate is set before the cluster is constructed`. However if that is the case, that is a concern of `CodegenIntegration` and that part should have workaround logic (i.e. save a pointer somewhere until cluster is constructed if it is not yet constructed). We should not let that leak inside the cluster class.